ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,314 @@
-  ## Insecure dependency in connect while running with -t switch at /usr/lib/perl5/5.8.8/i386-linux-thread-multi/IO/Socket.pm line 114.
-  ## Insecure dependency in connect while running with -t switch at /usr/lib/perl5/5.8.8/i386-linux-thread-multi/IO/Socket.pm line 120.
-
-  my $ugly_hack;
-  if ($master_cellservdb =~ m;(http://consult.cern.ch/service/afs/CellServDB);) {
-      $ugly_hack = 'http://consult.cern.ch/service/afs/CellServDB';
-      $master_cellservdb = $ugly_hack;
-  }
-  # end of UGLY HACK.
-
-  my $cellservdb_content = LWP::Simple::get($master_cellservdb);
-  if (! $cellservdb_content) {
-    $self->info("cannot read cellservdb $master_cellservdb, AFS cell info not changed");
     return 1;
-  }
-
-  if ($NoAction) {
-    $self->info("skipping CellServDB check and in-kernel update");
-  } else {
-    LC::Check::file("CellServDB",
-		    "contents"    => $cellservdb_content,
-		    "destination" => $localcelldb,
-		    "owner"       => 0,
-		    "mode"        => 0444,
-		   );
-    $self->update_afs_cells();
-  }
-
-  return 1;
 }
 
-# This code comes from sue/feature/afs/update.pl, slightly modified
-# to work in our context (replace afsutil with "fs")
-#
 # update the list of known AFS cells and run "fs newcell" when needed
 # This will only add new cells, not remove currently known ones.
 #
 sub update_afs_cells ( $$ ) {
-    my($self) = @_;
-    my($cell, %seen, %ipaddrs, @hosts, $todo, @todo, $error, $afsutil);
-    local(*FH, $_);
-    # init
-
-    # read CellServDB
-    # take it from the real file
-    if( ! open(*FH, $localcelldb)) {
-      $self->warn("cannot read local $localcelldb, giving up");
-      return 0;
-    }
+    my ($self) = @_;
+    my ( $cell, %seen, %ipaddrs, @hosts, $todo, @todo, $error, $afsutil );
 
-    while (<FH>) {
+    my $localcelldb_fh = CAF::FileEditor->new( $localcelldb, log => $self );
+    for ( split ( /\n/, "$localcelldb_fh" ) ) {
         if (/^>(.\S+)/) {
-            # new cell
             $cell = $1;
             $ipaddrs{$cell} = [];
-        } elsif (/^(\S+)\s+\#\s*(\S+)/) {
+        }
+        elsif (/^(\S+)\s+\#\s*(\S+)/) {
+
             # new entry
-            push(@{$ipaddrs{$cell}}, $1);
+            push ( @{ $ipaddrs{$cell} }, $1 );
         }
     }
-    close(FH);
-    foreach $cell (keys(%ipaddrs)) {
-        @{$ipaddrs{$cell}} = sort(@{$ipaddrs{$cell}});
-        $self->Debug("CellServDB $cell -> @{$ipaddrs{$cell}}");
+    $localcelldb_fh->close();
+    foreach $cell ( keys(%ipaddrs) ) {
+        @{ $ipaddrs{$cell} } = sort( @{ $ipaddrs{$cell} } );
+        $self->debug( 1, "CellServDB $cell -> @{$ipaddrs{$cell}}" );
     }
+
     # read known cells
     @todo = ();
-    my @current = `fs listcell -numeric 2>&1`;
+    my $proc = CAF::Process->new( [ "fs", "listcell", "-numeric" ], log => $self, keeps_state => 1 );
 
-    if($? >> 8 ) {
-      $self->info("cannot read current AFS cell info via fs:\n".join('',@current));
-      return 1;
+    if ( $? >> 8 ) {
```

Comment:
don't chekc exitcode before output. creating the `CAF::Process` instance does not run anything.